src/example-types/facet.js

```diff
@@ -38,6 +63,13 @@ module.exports = {
       ]),
     ]).then(([options, cardinality]) => ({
       cardinality: _.get('0.count', cardinality),
-      options: _.map(x => ({ name: x._id, count: x.count }), options),
+      options: _.map(
+        ({ _id, labelData, count }) => ({
+          name: _id,
+          ...(_.get('lookup', node) ? { labelData: labelData || {} } : {}),
```

Comment:
I don't think you need all these checks. `{ ...undefined }` will just give you `{}`

Reply:
@stellarhoof yeah, I just didn't want to junk up the object for people who didn't ask for this prop, that's why I check for node.lookup

The other issue is the `|| {}` -- my thought here was that if the lookup fails consumers won't have to guard against the nonexistence of `labelData` due to an expected record being missing from the foreign collection. But maybe that's a concern for the consumers.

Reply:
actually, nevermind, I just realized _.pick gives me this anyway. will update.

Reply:
scratch that,  I'm not picking this stuff -- will look into it

Reply:
yeah, since the fields are mapped into the keys of a $project, you're just not going to have `labelData` on the objects coming out of the mongo aggregation. So the question is: should we leave it to clients to guard against `labelData` not being here? I was thinking yes, but I can see the other view, too

Reply:
I agree with you on not having `labelData` if the client didn't ask for it. But I think this can be simplified still. Something like `F.compactObject({ name: _id, label, count })`

Reply:
👍 nice, I'll use that